test/python/test_transpiler.py

```diff
@@ -100,3 +101,96 @@ def test_pass_cx_cancellation(self):
         resources_after = dag_circuit.count_ops()
 
         self.assertNotIn('cx', resources_after)
+
+    def test_lookahead_mapper_doesnt_modify_mapped_circuit(self):
+        """Test that lookahead mapper is idempotent.
+
+        It should not modify a circuit which is already compatible with the
+        coupling map, and can applied repeatedly without modifying the circuit.
+        """
+
+        qr = QuantumRegister(3)
+        circuit = QuantumCircuit(qr)
+        circuit.cx(qr[0], qr[2])
+        circuit.cx(qr[0], qr[1])
+        original_dag = DAGCircuit.fromQuantumCircuit(circuit)
+
+        # Create coupling map which contains all two-qubit gates in the circuit.
+        coupling_map = Coupling({0: [1, 2]}, regname=qr.name)
```

Comment:
this is the old style coupling map. I think this should use the new style. Also we should discuss what coupling map is going to be verse gate definitions.